src/components/fields/date-input/date-input.tsx

```diff
@@ -114,23 +105,36 @@ export const DateInput = (props: IGenericFieldProps<IDateInputSchema>) => {
 			onChange({
 				target: { value: undefined },
 			});
-		} else if (!day.length || !month.length || year.length !== 4) {
+		} else if (day.length < 2 || month.length < 2 || year.length < 4) {
 			onChange({
 				target: { value: ERROR_MESSAGES.DATE.INVALID },
 			});
 		} else {
-			onChange({
-				target: {
-					value: DateHelper.formatDateTime(
-						[year, month.padStart(2, "0"), day.padStart(2, "0")].join("-"),
-						dateFormatter,
-						true
-					),
-				},
-			});
+			try {
+				onChange({
+					target: {
+						value: DateTimeHelper.formatDateTime([year, month, day].join("-"), dateFormat, "date"),
+					},
+				});
+			} catch (error) {
+				onChange({
+					target: { value: ERROR_MESSAGES.DATE.INVALID },
+				});
+			}
 		}
 	};
 
+	// =============================================================================
+	// HELPER FUNCTIONS
+	// =============================================================================
+	const isValidDate = (value: string) => {
+		if (!value) return false;
```

Comment:
can we change this to `if (!value || value === "" || value === ERROR_MESSAGES.DATE.INVALID)`